src/Config/RuntimeConfigLoader.cs

```diff
@@ -31,6 +31,12 @@ public abstract class RuntimeConfigLoader
     // state in place of using out params.
     public RuntimeConfig? RuntimeConfig;
 
+    public RuntimeConfig? lastValidRuntimeConfig;
+
+    public bool isNewConfigDetected;
+
+    public bool isNewConfigValidated = true;
```

Comment:
sorry not sure why both vars were highlighted, I meant `public bool isNewConfigValidated = true;` should `isNewConfigValidated ` be defaulted to false? e.g. a config is invalid until proven otherwise.